src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -208,37 +243,36 @@ private async Task ApplyContext(HostAssignmentContext assignmentContext)
             var options = _optionsFactory.Create(ScriptApplicationHostOptionsSetup.SkipPlaceholder);
             RunFromPackageContext pkgContext = assignmentContext.GetRunFromPkgContext();
 
-            var zipPath = pkgContext.Url;
-            if (!string.IsNullOrEmpty(zipPath))
+            if (!string.IsNullOrEmpty(pkgContext.Url) && pkgContext.Url != "1")
             {
-                // download zip and extract
-                var zipUri = new Uri(zipPath);
-                string filePath = null;
+                await ApplyBlobPackageContext(pkgContext.Url, options.ScriptPath);
+            }
+            else if (!string.IsNullOrEmpty(assignmentContext.AzureFilesConnectionString))
+            {
+                ApplyAzureFilesContext(assignmentContext.AzureFilesConnectionString, assignmentContext.AzureFilesContentShare, "/home");
+            }
+        }
 
-                if (pkgContext.IsScmRunFromPackage())
-                {
-                    bool blobExists = await pkgContext.BlobExistsAsync(_logger);
-                    if (blobExists)
-                    {
-                        filePath = await DownloadAsync(zipUri);
-                    }
-                    else
-                    {
-                        return;
-                    }
-                }
-                else
-                {
-                    filePath = await DownloadAsync(zipUri);
-                }
+        private void ApplyAzureFilesContext(string connectionString, string contentShare, string targetPath)
+        {
+            var sa = CloudStorageAccount.Parse(connectionString);
+            var key = Convert.ToBase64String(sa.Credentials.ExportKey());
 
-                UnpackPackage(filePath, options.ScriptPath);
+            var mountCommand = $"mount -t cifs //{sa.FileEndpoint.Host}/{contentShare} {targetPath} -o vers=3.0,username={sa.Credentials.AccountName},password={key},dir_mode=0777,file_mode=0777,serverino";
+            RunBashCommand($"(mkdir -p {targetPath} || true) && ({mountCommand}) && (mkdir -p /home/site/wwwroot || true)", MetricEventNames.LinuxContainerSpecializationAzureFilesMount);
+        }
 
-                string bundlePath = Path.Combine(options.ScriptPath, "worker-bundle");
-                if (Directory.Exists(bundlePath))
-                {
-                    _logger.LogInformation($"Python worker bundle detected");
-                }
+        private async Task ApplyBlobPackageContext(string blobUrl, string targetPath)
+        {
+            // download zip and extract
+            var blobUri = new Uri(blobUrl);
+            var filePath = await DownloadAsync(blobUri);
```

Comment:
Inside **DownloadAsync**, blobUri may return 404 if SCM_RUN_FROM_PACKAGE is set but no content is provided in blobUri. In that case, **response.EnsureSuccessStatusCode()** will throw an exception. We may want to handle the exception here.

Reply:
oh right, because it'll start out with a 404 one, right?